File: src/main/java/dev/lonami/uniffidl/psi/UdlFile.java
package dev.lonami.uniffidl.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import dev.lonami.uniffidl.UdlFileType;
import dev.lonami.uniffidl.UdlLanguage;
import org.jetbrains.annotations.NotNull;

public class UdlFile extends PsiFileBase {
    public UdlFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, UdlLanguage.INSTANCE);
    }

    @Override
    public @NotNull FileType getFileType() {
        return UdlFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "UDL File";
    }
}

File: src/main/kotlin/dev/lonami/uniffidl/psi/UdlFile.kt
package dev.lonami.uniffidl.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import dev.lonami.uniffidl.UdlFileType
import dev.lonami.uniffidl.UdlLanguage

class UdlFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, UdlLanguage.INSTANCE) {
    override fun getFileType(): FileType = UdlFileType.INSTANCE

    override fun toString(): String = "UDL File"
}
